filebeat/harvester/reader/strip_newline.go

```diff
@@ -20,6 +22,10 @@ func (p *StripNewline) Next() (Message, error) {
 
 	L := message.Content
 	message.Content = L[:len(L)-lineEndingChars(L)]
+
+	// Strip BOM
+	message.Content = bytes.Trim(message.Content, "\xef\xbb\xbf")
```

Comment:
the BOM sequence can only be at start of file/stream. Should/Must not in between. What about utf-32 bom?